### PR TITLE
OD-1342/no product id import

### PR DESCRIPTION
**Pull request by MPishpecki**

### CODE REVIEW

**Comment by MPishpecki** [comment]
@StefanRijnhart my PR is based on OCA 12.0. Our repository is behind the OCA. For this to be clean PR I have to fetch upstream.

File: connector_magento/models/sale_order/importer.py [hunk 1]
```diff
@@ -715,8 +715,9 @@ def _import_dependencies(self):
                     key = 'product_id'
                 else:
                     key = 'sku'
-                self._import_dependency(line[key],
-                                        'magento.product.product')
+            else:
+                key = 'sku'
+            self._import_dependency(line[key], 'magento.product.product')
```
Comment (StefanRijnhart) [inline connector_magento/models/sale_order/importer.py]: Please use this logic instead (from lower down this file): https://github.com/OCA/connector-magento/blob/12.0/connector_magento/models/sale_order/importer.py#L750-L753 and change the check to `if key in line`.

**Comment by StefanRijnhart** [comment]
Thanks! I fetched upstream on the target branch. Maybe you should rebase this one.

**Comment by StefanRijnhart** [comment]
Thanks. We need this in the upstream connector as well, so can you change the commit message according to OCA standards:
`[FIX] connector_magento: check right key depending on Magento version` and also push this to a public branch, and open a public PR.

**Comment by MPishpecki** [comment]
The OCA PR https://github.com/OCA/connector-magento/pull/330 
@astirpe Please have a look too. Thx.

**Review by astirpe — APPROVED** [review]

**Review by StefanRijnhart — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
Closing in favour of https://github.com/OCA/connector-magento/pull/330

**Pull request closed without merging**